2199/3/controllers/application_controller.rb

```diff
@@ -0,0 +1,38 @@
+require 'sinatra'
+require 'pry'
+# class ApplicationController
+class ApplicationController < Sinatra::Base
+  set views: 'views/'
+
+  get '/' do
+    redirect '/pages'
+  end
+
+  get '/pages' do
+    @pages = Page.all
+    erb :index
+  end
+
+  get '/pages/new' do
+    erb :new
+  end
+
+  get '/pages/:id' do
+    @page = Page[params[:id]]
+    erb :show
+  end
+
+  post '/pages' do
```

Comment:
Let's meditate on this endpoint.

I'd like to have less code in here.
Something like this:
```ruby
#app.rb
post '/pages' do
   if Page.exists?(link: link)
     @page = Page.find(link: link)
   else
     @page = Page.create_from(link: link)
   end
   erb :show
end
```
It's just an idea, but it will make your post action more verbose and will hide all Page and Comment creation logic.